src/clients/SpokePoolClient.ts

```diff
@@ -278,7 +278,7 @@ export class SpokePoolClient extends BaseAbstractClient {
    * @returns The corresponding SlowFIllRequest event if found, otherwise undefined.
    */
   public getSlowFillRequest(relayData: RelayData): SlowFillRequestWithBlock | undefined {
-    const hash = getRelayDataHash(relayData, this.chainId);
+    const hash = this.getDepositHash(relayData);
```

Comment:
Don't we lose resiliency to events that have the same deposit id now? Since getRelayDataHash hashes the entire struct. 

On the other hand, it shouldn't be possible to send duplicate slow fill requests

Reply:
It's not possible to make multiple slow fill requests for the same relayData hash, but it is still possible for the same depositId with different relayData hash to collide here. I agree that this change introduces a lack of collision-resistance that was not there before. However, we anyway have to solve this for our method of storing deposits within the SpokePoolClient, so this lack of resiliency was already there to begin with. My thinking was that we'd revisit `SpokePoolClient.getDepositHash()` and make it more resilient.

Reply:
could we not just hash the entire relay data but replacing the message with a message hash for deposits?

Reply:
Possibly, but we could also just borrow your approach from the BundleDataClient and concatenate all the fields (including messageHash instead of message). We probably want to move away from calling it a hash; maybe `getDepositKey()` or something?

We fwiw have a similar issue in the `queryHistoricalDeposit...()` function, where we're also relying on real relayData hashes, but in practice we only need a unique key.

Reply:
It's hard to prove, but my gut feel on the hash speed issue is that it's actually the abi encoding that's expensive, not the actual hashing. The ABI has to be instantiated on each each call to the ABICoder and it's supplied as stringified JSON. This is converted into an ABI instance on the fly. Doing this once per relayData seems to add a lot of overhead.

viem seems to support instantiating the ABI instance separately from encoding and hashing the data. I've got an old branch where I switched over to using it but I hit a snag on that. I don't recall what it was but I'd like to pick it up again because unlocking hashing inefficiencies has historically been a source of big speedups in the bots.